assets/artifact-hub-pkg/package.go

```diff
@@ -0,0 +1,296 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"bytes"
+	"io/ioutil"
+	"strings"
+	"net/http"
+	"time"
+	"path/filepath"
+
+	"gopkg.in/yaml.v3"
+
+	"github.com/layer5io/meshkit/utils"
+	meshkitErrors "github.com/layer5io/meshkit/errors"
+	"github.com/layer5io/meshkit/logger"
+
+	"github.com/layer5io/meshkit/utils/catalog"
+	"github.com/layer5io/meshkit/models/catalog/v1alpha1"
+)
+
+type CatalogPattern struct {
+	ID           string `json:"id"`
+	Name         string `json:"name"`
+	Version      string `json:"version"`
+	PatternFile  string `json:"pattern_file"`
+	CatalogData  v1alpha1.CatalogData `json:"catalog_data"`
+	UserID string `json:"user_id"`
+}
+
+type UserInfo struct {
+	UserID    string `json:"user_id"`
+	FirstName string `json:"first_name"`
+	LastName  string `json:"last_name"`
+	AvatarURL string `json:"avatar_url"`
+}
+
+const (
+	mesheryCloudBaseURL    = "https://meshery.layer5.io"
+	mesheryCatalogFilesDir = "catalog"
+)
+
+var(
+	ErrUnmarshalCatalogPatternCode = "test_code"
+	ErrProcessPatternCode          = "test_code"
+	ErrHTTPGetRequestCode          = "test_code"
+	ErrReadRespBodyCode            = "test_code"
+	ErrCreateGitHubRequestCode     = "test_code"
+    ErrInvokeGitHubActionsCode     = "test_code"
+)
+
+func main() {
+	token := os.Getenv("GH_ACCESS_TOKEN")
+	log, err := logger.New("mesheryio_package", logger.Options{
+		Format:     logger.SyslogLogFormat,
+		LogLevel:   5,
+	})
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	catalogPatterns, err := fetchCatalogPatterns()
+	if err != nil {
+		log.Error(err)
+		return
+	}
+
+	var patterns struct {
+		Patterns []CatalogPattern `json:"patterns"`
+	}
+	if err := json.Unmarshal(catalogPatterns, &patterns); err != nil {	
+		log.Error(utils.ErrUnmarshal(err))
+		return 
+	}
+
+	for _, pattern := range patterns.Patterns {
+		if err := processPattern(pattern, token); err != nil {
+			log.Error(meshkitErrors.New(ErrProcessPatternCode, meshkitErrors.Alert,
```

Comment:
any feedback here, @Jougan-0?

Reply:
IMO you can define this Meshkit error function the same as others at bottom of the file.